controller/clusters.go

```diff
@@ -92,6 +92,61 @@ func (c *ClustersController) ShowAuthClient(ctx *app.ShowAuthClientClustersConte
 	return ctx.OK(&clusters)
 }
 
+// Show returns the single clusters.
+func (c *ClustersController) Show(ctx *app.ShowClustersContext) error {
+	if !auth.IsSpecificServiceAccount(ctx, auth.OsoProxy, auth.Tenant, auth.JenkinsIdler, auth.JenkinsProxy, auth.Auth) {
+		log.Error(ctx, nil, "unauthorized access to cluster info")
+		return app.JSONErrorResponse(ctx, errors.NewUnauthorizedError("unauthorized access to cluster info"))
+	}
+	clustr, err := c.app.ClusterService().Load(ctx, ctx.ClusterID)
+	if err != nil {
+		return app.JSONErrorResponse(ctx, err)
+	}
+	return ctx.OK(&app.ClusterSingle{
+		Data: &app.ClusterData{
+			Name:              clustr.Name,
+			APIURL:            httpsupport.AddTrailingSlashToURL(clustr.URL),
+			ConsoleURL:        httpsupport.AddTrailingSlashToURL(clustr.ConsoleURL),
+			MetricsURL:        httpsupport.AddTrailingSlashToURL(clustr.MetricsURL),
+			LoggingURL:        httpsupport.AddTrailingSlashToURL(clustr.LoggingURL),
+			AppDNS:            clustr.AppDNS,
+			Type:              clustr.Type,
+			CapacityExhausted: clustr.CapacityExhausted,
+		},
+	})
+}
+
+// ShowForAuthClient returns the cluster with full configuration including Auth client data.
+// To be used by Auth service only
+func (c *ClustersController) ShowForAuthClient(ctx *app.ShowForAuthClientClustersContext) error {
+	if !auth.IsSpecificServiceAccount(ctx, auth.Auth) {
+		log.Error(ctx, nil, "unauthorized access to cluster info")
+		return app.JSONErrorResponse(ctx, errors.NewUnauthorizedError("unauthorized access to cluster info"))
+	}
+	clustr, err := c.app.ClusterService().Load(ctx, ctx.ClusterID)
+	if err != nil {
+		return app.JSONErrorResponse(ctx, err)
+	}
+	return ctx.OK(&app.FullClusterSingle{
+		Data: &app.FullClusterData{
```

Comment:
As we already have this defined `ListForAuthClient`. I think we can move it one function and reuse it from there to follow DRY principle

Reply:
same response as above.